mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -428,6 +428,51 @@ static LogicalResult verifyConvOpModes(T op) {
   return success();
 }
 
+// Verify whether same length and type of block arguments and tensor list.
+static LogicalResult errorIfTensorListShapeMismatch(Operation *op,
+                                                    ValueRange blocksArgs,
+                                                    StringRef blockName,
+                                                    ValueRange tensorList,
+                                                    StringRef listName) {
+  if (blocksArgs.size() != tensorList.size())
+    return op->emitOpError() << "require same number of values in " << blockName
+                             << " (" << blocksArgs.size() << ") and "
+                             << listName << " (" << tensorList.size() << ")";
+
+  for (auto [bbArgType, opArgType] :
+       llvm::zip_equal(blocksArgs.getTypes(), tensorList.getTypes())) {
+    ShapeAdaptor bbShapeAdaptor(bbArgType);
+    ShapeAdaptor opShapeAdaptor(opArgType);
+
+    if (!bbShapeAdaptor.hasRank() || !opShapeAdaptor.hasRank())
+      continue;
+
+    if (!bbShapeAdaptor.hasStaticShape() || !opShapeAdaptor.hasStaticShape())
```

Comment:
I think we could make this check a bit more strict, e.g. comparison of `tensor<1x?x3xf32>` and `tensor<1x2x2xf32>` would return success even though we can prove it to be a failure

Reply:
Nice suggestion.